spec.bs

```diff
@@ -972,14 +751,351 @@ To <dfn>generate reporting URLs</dfn> given an {{AuctionAdConfig}} |config| and
+  |functionName|, and a [=list=] |arguments|, run these steps. They return a [=ECMAScript/Completion
+  Record=], which is either an [=abrupt completion=] (in the case of a parse failure or execution
+  error), or a normal completion populated with the [=ECMAScript/ECMAScript language value=] result
+  of invoking |functionName|.
+
+    1. [=Assert=] that these steps are running [=in parallel=].
+
+    1. Let |agent| be the result of [=obtaining a script runner agent=] given null, true, and
+       false. Run the rest of these steps in |agent|.
+
+       Issue: This exclusively creates a new [=ECMAScript/agent cluster=] for the given |script| to
+       run in, but we should make this work with [=interest group/execution mode=] somehow.
+
+    1. Let |realmExecutionContext| be the result of [=creating a new realm=] given |agent| and the
+       following customizations:
+
+        * For the global object, create a new object of type |globalType|.
+
+    1. Let |realm| be |realmExecutionContext|'s Realm component.
+
+    1. Let |global| be |realm|'s [=realm/global object=], and run these steps:
+
+      1. Perform !|global|.\[[Delete]]("`Date`").
+
+      1. If !|global|.\[[HasProperty]]("`Temporal`") is true, then perform
+         !|global|.\[[Delete]]("`Temporal`").
+
+      Advisement: This is not the best way to perform such API neutering (see <a
+      href=https://github.com/tc39/ecma262/issues/1357#issuecomment-817560121>tc39/ecma262#1357</a>),
+      but at the moment it's the way that host environments do this.
+
+      Note: Removing time-referencing APIs from the |global| object is imperative for privacy, as
+      |script| would otherwise be able to accurately exfiltrate data gleaned from time measurements.
+
+    1. Let |result| be [=ParseScript=](|script|, |realm|, `empty`).
+
+       Note: The resulting [=ECMAScript/Script Record=] will have no \[[HostDefined]] component,
+       unlike traditional [=scripts=] on the web platform.
+
+    1. If |result| is a list of errors, return
+       Completion { \[[Type]]: `throw`, \[[Value]]: |result|, \[[Target]]: `empty` }.
+
+    1. [=Assert=]: |result| is a [=ECMAScript/Script Record=].
+
+    1. *Prepare to run script*: Push |realmExecutionContext| onto the [=ECMAScript/execution context
+       stack|JavaScript execution context stack=]; it is now the [=ECMAScript/running execution
+       context|running JavaScript execution context=].
+
+    1. Let |evaluationStatus| be the result of [=ScriptEvaluation=](result).
+
+    1. If |evaluationStatus| is an [=abrupt completion=], jump to the step labeled <i><a
+       href=#evaluate-script-return>return</a></i>.
+
+    1. Let |F| be [$Get$](|global|, |functionName|). If that returns a [=throw completion=], set
+       |finalCompletion| to |F| and jump to the step labeled <i><a
+       href=#evaluate-script-return>return</a></i>.
+
+    1. Set |finalCompletion| be [=ECMAScript/Completion Record|Completion=]([$Call$](F, `undefined`,

```

Comment:
Oh whoah, that's super odd... there's not much precedent for this on the web. Does Chromium just conditionally inject the "global" functions depending on what function is entered or something? What is the benefit of this? Just making sure that the `setBid()` function is not called when the overall script is "run" _before_ the user's global functions are entered?

Reply:
The more webby way to do this would probably be to make the `generateBid()` function take a parameter (a "bid" I guess?) that the `generateBid()` function body fills out. That way the only way you get access to the bid-that-you-have-to-fill-out-or-generate from within `generateBid()`, which sounds like your intention.